src/common/steps/verify/Verification.tsx

```diff
@@ -0,0 +1,183 @@
+    getResponses,
+    getShowSkip,
+    setFailed,
+    setResponses,
+} from './verificationSlice';
+
+export interface Command {
+    title: string;
+    command: string;
+    responseRegex: string;
+    copiable?: boolean;
+}
+
+const formatResponse = (response: string, responseRegex: RegExp) => {
+    const filteredResponse = response
+        .split('\n')
+        .filter(line => !!line.trim() && line.trim() !== 'OK')
+        .join('')
+        .trim();
+
+    const [, match] = filteredResponse.match(responseRegex) ?? [];
+
+    return match;
+};
+
+const runVerification =
+    (
+        commands: Command[],
+        device: DeviceWithSerialnumber
+    ): AppThunk<RootState, Promise<void>> =>
+    async dispatch => {
+        dispatch(setFailed(false));
+        let serialPort: Awaited<ReturnType<typeof getUARTSerialPort>>;
+        try {
+            serialPort = await getUARTSerialPort(device);
+        } catch (error) {
+            logger.error(describeError(error));
+            dispatch(setFailed(true));
+            return;
+        }
+
+        const newResponses: string[] = [];
+        const reducedPromise = commands.reduce(
+            (acc, next) =>
+                acc.then(() =>
+                    serialPort.sendCommand(next.command).then(value => {
+                        newResponses.push(
+                            formatResponse(
+                                value,
+                                new RegExp(next.responseRegex)
+                            )
+                        );
+
+                        return Promise.resolve();
+                    })
+                ),
+            Promise.resolve()
+        );
+        try {
+            await reducedPromise;
+
+            dispatch(setResponses(newResponses));
+            serialPort.unregister();
+        } catch (e) {
+            logger.error('Received ERROR as return value from AT command');
+            dispatch(setFailed(true));
+        }
+    };
+
+export default ({ commands }: { commands: Command[] }) => {
+    const dispatch = useAppDispatch();
+    const device = useAppSelector(getSelectedDeviceUnsafely);
+    const responses = useAppSelector(getResponses);
+    const failed = useAppSelector(getFailed);
+    const showSkip = useAppSelector(getShowSkip);
+    const [verifying, setVerifying] = useState(false);
+
+    const gotAllResponses = responses.every(response => response !== '');
+
+    const getHeading = () => {
+        if (!verifying) {
+            return 'Verify';
+        }
+        if (failed) {
+            return 'Verification failed';
+        }
+        if (gotAllResponses) {
+            return 'Verification successful';
+        }
+        return 'Verifying';
+    };
+
+    return (
+        <Main>
+            <Main.Content
+                heading={getHeading()}
+                subHeading={
+                    !verifying
+                        ? 'Automatically verify kit communication with AT commands.'
```

Comment:
Minor change in perspective, but relevant for the given scenario:
```suggestion
                        ? 'Automatically verify kit's communication with AT commands.'
```